src/tir/schedule/primitive/blockize_tensorize.cc

```diff
@@ -0,0 +1,638 @@
+  ICHECK(buffer_region->region.size() == buffer_region->buffer->shape.size());
+  for (size_t i = 0; i < buffer_region->region.size(); i++) {
+    Range max_range = Range::FromMinExtent(0, buffer_region->buffer->shape[i]);
+    new_region.push_back(relaxed_int_set[i].CoverRange(max_range));
+  }
+  return BufferRegion(buffer_region->buffer, std::move(new_region));
+}
+
+/*!
+ * \brief Generate the outer block after blockize.
+ * \param extractor The binding extractor which has extracted the blockized bindings.
+ * \param block The original inner block.
+ * \param inner_block_realize The block realize of the inner block after blockize.
+ * \param inner_loops The inner loops after blockize.
+ * \param predicate The outer predicate of the subspace division.
+ * \param analyzer The arithmetic analyzer.
+ * \return The block realize of the outer block after blockize.
+ */
+BlockRealize GenerateBlockizedOuterBlock(const BlockizedBindingExtractor& extractor,
+                                         const Block& block, BlockRealize inner_block_realize,
+                                         const std::vector<const ForNode*>& inner_loops,
+                                         PrimExpr predicate, arith::Analyzer* analyzer) {
+  // Step 1: Generate the init block if needed
+  Optional<Stmt> new_init = NullOpt;
+  if (block->init.defined()) {
+    new_init = GenerateBlockizedInit(block, inner_block_realize, inner_loops);
+  }
+
+  // Step 2: Compute the access regions of the outer block by relaxing the inner loops
+  Array<BufferRegion> new_reads = block->reads;
+  Array<BufferRegion> new_writes = block->writes;
+
+  auto f_mutate = [&](const BufferRegion& buffer_region) {
+    return RelaxBlockizedInnerIters(buffer_region, extractor.inner_iter_relaxed_range, analyzer);
+  };
+  new_reads.MutateByApply(f_mutate);
+  new_writes.MutateByApply(f_mutate);
+
+  // Step 3: Generate the body of the outer block. The body of the outer block is the inner block
+  // realize and its surounding loops.
+  Stmt outer_block_body = inner_block_realize;
+  for (const ForNode* loop : inner_loops) {
+    ObjectPtr<ForNode> new_loop = make_object<ForNode>(*loop);
+    new_loop->body = std::move(outer_block_body);
+    outer_block_body = For(new_loop);
+  }
+
+  // Step 4: Generate the outer block and block realize.
+  Block outer_block{/*iter_vars=*/extractor.outer_iter_vars,        //
+                    /*reads=*/new_reads,                            //
+                    /*writes=*/new_writes,                          //
+                    /*name_hint=*/"blockized_" + block->name_hint,  //
+                    /*body=*/std::move(outer_block_body),           //
+                    /*init=*/new_init};
+  BlockRealize outer_block_realize{/*iter_values=*/extractor.outer_bindings,
+                                   /*predicate=*/std::move(predicate),
+                                   /*block=*/std::move(outer_block)};
+  return outer_block_realize;
```

Comment:
nit: Let's make the logic more concise here

```suggestion
  return BlockRealize(/*iter_values=*/std::move(extractor.outer_bindings),
                      /*predicate=*/std::move(predicate),
                      /*block=*/
                      Block(/*iter_vars=*/std::move(extractor.outer_iter_vars),  //
                            /*reads=*/std::move(new_reads),                      //
                            /*writes=*/std::move(new_writes),                    //
                            /*name_hint=*/block->name_hint + "_o",               //
                            /*body=*/std::move(outer_block_body),                //
                            /*init=*/std::move(new_init)));
```